Packs/CloudConvert/Integrations/CloudConvert/CloudConvert.py

```diff
@@ -0,0 +1,369 @@
+        :rtype: ``Dict[str, Any]``
+        """
+        return self._http_request(
+            method='POST',
+            url_suffix='import/url',
+            headers=self._headers,
+            data=arguments,
+            timeout=25,
+            ok_codes=(422, 200, 201, 500, 401)
+        )
+
+    def import_entry_id(self, file_path: str, file_name: str) -> Dict[str, Any]:
+        """
+        Import the file given as a war room entry id to the API's server, for later conversion
+
+        :param file_path: path to given file, derived from the entry id
+        :param file_name: name of file, including format suffix
+        :return: dict containing the results of the import action as returned from the API (status, file ID, etc.)
+        :rtype: ``Dict[str, Any]``
+        """
+        response_get_form = self._http_request(
+            method='POST',
+            url_suffix='import/upload',
+            headers=self._headers
+        )
+        form = response_get_form.get('data').get('result').get('form')
+        port_url = form.get('url')
+        params = form.get('parameters')
+
+        # Creating a temp file with the same data of the given file
+        # This way the uploaded file has a path that the API can parse properly
+        demisto.debug('creating a temp file for upload operation')
+        with tempfile.TemporaryFile(suffix=file_name) as temp_file:
+            with open(file_path, 'rb') as file:
+                temp_file.write(file.read())
+            temp_file.seek(0)
+            file_dict = {file_name: temp_file}
+
+            self._http_request(
+                method='POST',
+                url_suffix=None,
+                full_url=port_url,
+                headers=self._headers,
+                files=file_dict,
+                empty_valid_codes=[201, 204],
+                return_empty_response=True,
+                timeout=25,
+                data=params
+            )
+
+        response_get_form['data'].pop('message')
+        response_get_form['data'].pop('result')
+        return response_get_form
+
+    def convert(self, arguments: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Convert a file to desired format, given the file was priorly imported to the API's server
+
+        :type arguments: ``Dict[str, Any]``
+        :param arguments: dict containing the request arguments, should contain the fields 'task_id' and 'output_format'
+
+        :return: dict containing the results of the convert action as returned from the API (status, file ID, etc.)
+        :rtype: ``Dict[str, Any]``
+        """
+        arguments['input'] = arguments.pop('task_id')
+        return self._http_request(
+            method='POST',
+            url_suffix='convert',
+            headers=self._headers,
+            data=arguments,
+            timeout=25,
+            ok_codes=(422, 200, 201, 500)
```

Comment:
all of these code are potentially linked to things that the user can fix - such as wrong task ids, wrong output formats, wrong urls, etc. if this code is being raised as a result of one of those cases, the response json will have a field of 'data' and within 'message' which while describe the problem. in the command (the one wrapping this function) there is a check if the 'data' and 'message' fields are present - if they are present in the json, then an error with that specific message is raised (and therefore the user knows what he needs to fix). If there is no data field (and no message), meaning there is a different problem (possibly internal error like you mentioned) then an error is still raised, with a general message (see line 307). do you think this can still cause a problem?

Reply:
better to let the `500` result to raise an exception.